src/Emitter.ts

```diff
@@ -80,7 +80,12 @@ export class Emitter<Events extends EventMap> {
   ): Listener<Events[EventName]> {
     const onceListener = (...data: Events[keyof Events]) => {
       this.removeListener(eventName, onceListener)
-      listener.apply(this, data)
+
+      /**
+       * @note Return the result of the original listener.
+       * This way this wrapped preserves listeners that are async.
+       */
+      return listener.apply(this, data)
```

Comment:
By returning this listener call, this wrapped will respect the sync/async nature of the original listener, and can, itself, be awaited.